### PR TITLE
fix: [#2745] Fix current eslint warnings - botbuilder-dialogs-adaptive library - misc

### PR DESCRIPTION
**Pull request by NahuelPerales**
Adresses # 2745
#minor

## Description
When reviewing the botbuilder-js project it was found that eslint is showing several warnings for the botbuilder-dialogs-adaptive library.

## Specific Changes

- Replaced strings quotes with the correct one depending of the case
- Replaced hasOwnProperty with Object.prototype.hasOwnProperty.call
- Adjusted formatting
- Limited importing scopes
- Added missing jsdocs documentation for parameters and returns for functions
- Removed unused variables

## Testing
Eslint with no warnings and tests passed
image

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive/src/functions/hasPendingActionsFunction.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class HasPendingActionsFunction extends ExpressionEvaluator {
         super(HasPendingActionsFunction.functionName, HasPendingActionsFunction.function, ReturnType.Boolean);
     }
 
-    private static function(expression: Expression, state: MemoryInterface, options: Options): ValueWithError {
+    private static function(expression: Expression, state: MemoryInterface, _options: Options): ValueWithError {
```
Comment (JuanAr) [inline libraries/botbuilder-dialogs-adaptive/src/functions/hasPendingActionsFunction.ts:35]: I'm not sure why this param is renamed. There might be a good reason.

File: libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts [hunk 2]
```diff
@@ -279,13 +286,14 @@ export abstract class InputDialog extends Dialog implements InputDialogConfigura
...
      * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
-     * @param reason [DialogReason](xref:botbuilder-dialogs.DialogReason), reason why the dialog resumed.
-     * @param result Optional. Value returned from the [Dialog](xref:botbuilder-dialogs.Dialog) that was called.
+     * @param _reason [DialogReason](xref:botbuilder-dialogs.DialogReason), reason why the dialog resumed.
+     * @param _result Optional. Value returned from the [Dialog](xref:botbuilder-dialogs.Dialog) that was called.
      * The type of the value returned is dependent on the child dialog.
      * @returns A [DialogTurnResult](xref:botbuilder-dialogs.DialogTurnResult) `Promise` representing the asynchronous operation.
      */
-    public async resumeDialog(dc: DialogContext, reason: DialogReason, result?: any): Promise<DialogTurnResult> {
+    public async resumeDialog(dc: DialogContext, _reason: DialogReason, _result?: any): Promise<DialogTurnResult> {
```
Comment (JuanAr) [inline libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts:296]: I realized that, unused params are prefixed with underscore. Is this a ESLint rule?

File: libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts [hunk 3]
```diff
@@ -322,10 +330,11 @@ export abstract class InputDialog extends Dialog implements InputDialogConfigura
     /**
      * @protected
      * Method which processes options.
-     * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param _dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
      * @param options Initial information to pass to the dialog.
+     * @returns A promise representing the asynchronous operation.
      */
-    protected onInitializeOptions(dc: DialogContext, options: any): Promise<any> {
+    protected onInitializeOptions(_dc: DialogContext, options: any): Promise<any> {
```
Comment (JuanAr) [inline libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts:337]: same

File: libraries/botbuilder-dialogs-adaptive/src/input/oauthInput.ts [hunk 4]
```diff
@@ -333,10 +341,9 @@ export class OAuthInput extends InputDialog implements OAuthInputConfiguration {
     /**
      * @protected
      * Called when input has been received.
-     * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
-     * @returns [InputState](xref:botbuilder-dialogs-adaptive.InputState) which reflects whether input was recognized as valid or not.
+     * @param _dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
      */
-    protected onRecognizeInput(dc: DialogContext): Promise<InputState> {
+    protected onRecognizeInput(_dc: DialogContext): Promise<InputState> {
```
Comment (JuanAr) [inline libraries/botbuilder-dialogs-adaptive/src/input/oauthInput.ts:346]: same

**Review by JuanAr — COMMENTED** [review]
I left some comments, not request changes, because there might be good reasons to for those.

**Review by JuanAr — APPROVED** [review]
LGTM

File: libraries/botbuilder-dialogs-adaptive/src/selectors/mostSpecificSelector.ts [hunk 5]
```diff
@@ -14,19 +14,34 @@ export interface MostSpecificSelectorConfiguration {
     selector?: TriggerSelector;
 }
 
+/**
+ *
+ */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/selectors/mostSpecificSelector.ts:19]: same

File: libraries/botbuilder-dialogs-adaptive/src/templates/staticActivityTemplate.ts [hunk 6]
```diff
@@ -33,10 +34,18 @@ export class StaticActivityTemplate
...
+     * @returns The converter for the selector configuration.
+     */
     public getConverter(_property: keyof StaticActivityTemplateConfiguration): Converter | ConverterFactory {
         return undefined;
     }
 
+    /**
+     * @param config The configuration.
+     * @returns A object with the given configuration.
+     */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/templates/staticActivityTemplate.ts:48]: For this case i think is valid that doesnt have a description, it gets inherited from the Configurable abstract class, we should do the same for the getConverter method

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Comment by sw-joelmut** [comment]
Promoted `https://github.com/microsoft/botbuilder-js/pull/4068`

**Pull request closed without merging**